feature/experimental/system/gnmi/benchmarking/ate_tests/internal/setup/setup.go

```diff
@@ -0,0 +1,324 @@
+	"github.com/openconfig/featureprofiles/internal/deviations"
+	"github.com/openconfig/featureprofiles/internal/fptest"
+	"github.com/openconfig/ondatra"
+	"github.com/openconfig/ondatra/gnmi"
+	"github.com/openconfig/ondatra/gnmi/oc"
+	"github.com/openconfig/ygnmi/ygnmi"
+	"github.com/openconfig/ygot/ygot"
+)
+
+// Some of the variables defined below like DutAS, AteAS, PeerGrpName
+// RouteCount and IsisInstance are used in other files which import
+// setup.go
+const (
+	// DutAS can be exported
+	DutAS = 64500
+	// AteAs can be exported
+	AteAS  = 64501
+	ateAS2 = 64502
+	// PeerGrpName can be exported
+	PeerGrpName    = "BGP-PEER-GROUP"
+	plenIPv4       = 30
+	dutStartIPAddr = "192.0.2.1"
+	ateStartIPAddr = "192.0.2.2"
+	// RouteCount can be exported
+	RouteCount            = 200
+	advertiseBGPRoutesv4  = "203.0.113.1"
+	authPassword          = "ISISAuthPassword"
+	advertiseISISRoutesv4 = "198.18.0.0"
+	IsisInstance          = "DEFAULT"
+	ISISMetric            = 100
+	// DUTAreaAddress can be exported
+	DUTAreaAddress = "49.0001"
+	// DUTSysID can be exported
+	DUTSysID = "1920.0000.2001"
+)
+
+var (
+	// DutIPPool can be exported
+	DutIPPool = make(map[string]net.IP)
+	// AteIPPool can be exported
+	AteIPPool = make(map[string]net.IP)
+	// ISISMetricArray can be exported
+	ISISMetricArray []uint32
+	// ISISSetBitArray can be exported
+	ISISSetBitArray []bool
+)
+
+// BuildIPPool is to Build pool of ip addresses for both DUT and ATE interfaces.
+// It reads ports given in binding file to calculate ip addresses needed.
+func BuildIPPool(t *testing.T) {
+	dut := ondatra.DUT(t, "dut")
+	var dutIPIndex, ipSubnet, ateIPIndex int = 1, 2, 2
+	var endSubnetIndex = 253
+	for _, dp := range dut.Ports() {
+		dutNextIP := nextIP(net.ParseIP(dutStartIPAddr), dutIPIndex, ipSubnet)
+		ateNextIP := nextIP(net.ParseIP(ateStartIPAddr), ateIPIndex, ipSubnet)
+		DutIPPool[dp.ID()] = dutNextIP
+		AteIPPool[dp.ID()] = ateNextIP
+
+		// Increment dut and ate host ip index by 4
+		dutIPIndex = dutIPIndex + 4
+		ateIPIndex = ateIPIndex + 4
+
+		// Reset dut and ate IP indexes when it is greater endSubnetIndex
+		if dutIPIndex > int(endSubnetIndex) {
+			ipSubnet = ipSubnet + 1
+			dutIPIndex = 1
+			ateIPIndex = 2
+		}
+	}
+
+}
+
+// nextIP returns ip address based on hostindex and subnetindex provided.
+func nextIP(ip net.IP, hostIndex int, subnetIndex int) net.IP {
+	s := ip.String()
+	sa := strings.Split(s, ".")
+	sa[2] = strconv.Itoa(subnetIndex)
+	sa[3] = strconv.Itoa(hostIndex)
+	s = strings.Join(sa, ".")
+	return net.ParseIP(s)
+}
+
+// BuildOCUpdate function is to build  OC config for interfaces.
+// It reads ports from binding file and returns gpb update message
+// which will have configurations for all the ports.
+func BuildOCUpdate(t *testing.T) {
+	dut := ondatra.DUT(t, "dut")
+	d := &oc.Root{}
+
+	// Default Network instance and Global BGP configs
+	ni1 := d.GetOrCreateNetworkInstance(*deviations.DefaultNetworkInstance)
```

Comment:
Hmm, "ni1" looks a lot like "nil" to my eye. Maybe just me ...

Reply:
renamed